src/control/PlanarArmController.h

```diff
@@ -36,15 +36,59 @@ class PlanarArmController {
 	 * @param safetyFactor the percentage factor to scale maximum arm extension radius by to
 	 * prevent singularity lock.
 	 */
-	PlanarArmController(const navtypes::Vectord<N>& currJointPos,
-						kinematics::PlanarArmKinematics<N> kin_obj, const double safetyFactor)
-		: kin(kin_obj), velocity({0.0, 0.0}), safetyFactor(safetyFactor) {
-		// NOTE: currJointPos could extend beyond the safetyFactor, so safety factor
-		//       normalization logic is performed.
-		set_setpoint(currJointPos);
+	PlanarArmController(kinematics::PlanarArmKinematics<N> kin_obj, const double safetyFactor)
+		: kin(kin_obj), safetyFactor(safetyFactor) {
 		CHECK_F(safetyFactor > 0.0 && safetyFactor < 1.0);
 	}
 
+	/**
+	 * @brief Constructs a copy of an existing PlanarArmController object.
+	 *
+	 * @param other The existing PlanarArmController to copy.
+	 */
+	PlanarArmController(PlanarArmController&& other)
+		: kin(std::move(other.kin)), safetyFactor(other.safetyFactor) {
+		std::lock_guard<std::mutex> lock(other.mutex);
+		mutableFields = std::move(other.mutableFields);
+	}
+
+	/**
+	 * @brief Instantiates and returns an std::optional containing a PlanarArmController.
+	 *
+	 * @param currJointPos The current joint positions of the arm.
+	 * @param kin_obj PlanarArmKinematics object for the arm (should have the same number of
+	 * arm joints).
+	 * @param safetyFactor the percentage factor to scale maximum arm extension radius by to
+	 * prevent singularity lock.
+	 * @return an std::optional containing a PlanarArmController if the joint positions are
+	 * within the robot's maximum arm extension radius, or an empty std::optional otherwise.
+	 */
+	bool tryInitController(const navtypes::Vectord<N>& currJointPos) {
+		if (is_setpoint_valid(currJointPos, kin, safetyFactor)) {
+			set_setpoint(currJointPos);
```

Comment:
Don't we need to initialize `mutableFields` before we call `set_setpoint`? Otherwise we'll try accessing an empty optional. Remember to lock the mutex! In this case, it probably makes sense to initialize `mutableFields` and all its fields without calling `set_setpoint`, so we don't have to worry about double-locking of mutexes or anything.